File: DomainModels/Dto/UserDto/UserDto.cs
using DomainModels.Dto.UserProfileDto;

namespace DomainModels.Dto.UserDto;
public class UserDto
{
	public int Id { get; set; }
	public string Email { get; set; } 
	public string UserName { get; set; } 
	public string UserRole { get; set; } 
	public string HashedPasword { get; set; }
	public UserInfoGetDto? UserInfo { get; set; }
	public DateTime? LastLogin { get; set; }
	public DateTime CreatedAt { get; set; }
	public DateTime? UpdatedAt { get; set; }
}

File: dashboard/src/types/UserDTO.ts
// src/types/UserDTO.ts

export type UserInfoDTO = {
  userId: number
  firstName: string
  lastName: string
  address: string
  postalCode: string
  city: string
  country: string
  phoneNumber: string
  dateOfBirth: string
  specialRequests: string
  createdAt: string
  updatedAt: string
}

export type UserDTO = {
  id: number
  email: string
  userName: string
  userRole: string
  hashedPasword: string
  userInfo: UserInfoDTO
  lastLogin: string
  createdAt: string
  updatedAt: string
}
